conf/modules/filter_variants.config

```diff
@@ -0,0 +1,77 @@
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Config file for defining DSL2 per module options and publishing paths
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Available keys to override module options:
+        ext.args   = Additional arguments appended to command in module.
+        ext.args2  = Second set of arguments appended to command in module (multi-tool modules).
+        ext.args3  = Third set of arguments appended to command in module (multi-tool modules).
+        ext.prefix = File name prefix for output files.
+----------------------------------------------------------------------------------------
+*/
+
+process {
+
+    /*
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Filter variants
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    */
+
+    withName: '.*:FILTER_VARIANTS_SNV:.*' {
+        publishDir = [
+            enabled: false,
+        ]
+    }
+
+    withName: '.*:FILTER_VARIANTS_SNVS:ENSEMBLVEP_FILTERVEP' {
+        ext.args   = { "--filter \"HGNC_ID in ${feature_file}\"" }
+        publishDir = [
+            enabled: false,
+        ]
+    }
+
+    withName: '.*:FILTER_VARIANTS_SVS:ENSEMBLVEP_FILTERVEP' {
+        ext.args   = { "--filter \"HGNC_ID in ${feature_file}\"" }
+        publishDir = [
+            enabled: false,
+        ]
+    }
+
+    withName: '.*:FILTER_VARIANTS_SNVS:BCFTOOLS_VIEW' {
+        ext.prefix = { params.skip_snv_annotation ? "${meta.id}_snvs_filtered" : (params.skip_rank_variants ? "${meta.id}_snvs_annotated_filtered" : "${meta.id}_snvs_annotated_ranked_filtered") }
+        ext.args   = { [
+            '--output-type z',
+            '--write-index=tbi',
+            "${params.filter_snvs_expression}"
+        ].join(" ") }
+        publishDir = [
+            path: { "${params.outdir}/snvs/multi_sample/${meta.id}" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
+    }
+
+    withName: '.*:FILTER_VARIANTS_SVS:BCFTOOLS_VIEW' {
+        ext.prefix = {
+            def parts = []
+            parts << "${meta.id}"
+            parts << (params.skip_cnv_calling ? 'svs_merged' : 'svs_cnvs_merged')
+            if (!params.skip_sv_annotation) parts << 'annotated'
+            if (!params.skip_rank_variants) parts << 'ranked'
+            parts << 'filtered'
+            return parts.join('_')
```

Comment:
Is it possible to use the function simpleName instead? Something like this 
`ext.prefix = {  ${input.simpleName}_filtered" }`

Reply:
It would be simpler.

For this workflow it would be possible since the filtering is done on the merged files (as opposed to chromosome regions), but it would break if the sample name contains dots. 

Perhaps we should restrict the input sample names to characters without dots if we want to use simpleName (same for raredisease).

Reply:
Yeah dots in sample names would break this. I think it would be worth looking into restricting dots in sample names in order to simplify the setting of prefixes. Not urgent though and in a future PR